### DISCUSSION TITLE
Help with linkerd-identity env variable

### DISCUSSION THREAD

**Discussion by rnsv** [linkerd/linkerd2#5448] [Q&A]
When using proxy-injector webhook (linkerd.io/inject: enabled) in linkerd-identity to insert linkerd-proxy container, the env value LINKERD2_PROXY_IDENTITY_SVC_ADDR is being injected as
```
    - name: LINKERD2_PROXY_IDENTITY_SVC_ADDR
      value: linkerd-identity-headless.linkerd.svc.cluster.local:8080
```
instead of 
```
    - name: LINKERD2_PROXY_IDENTITY_SVC_ADDR
      value: localhost.:8080
```
This results in health check failing with error -  Readiness probe failed: HTTP probe failed with statuscode: 503

I am using version 2.9.1.  This used to properly work in 2.7.1

What I am looking for is to mimic this behavior https://github.com/linkerd/linkerd2/blob/stable-2.9.1/charts/partials/templates/_proxy.tpl#L81 when using proxy-injector webhook for linkerd-identity

Thank you

**Comment by cpretzer** [comment]
Looks like there was a [change](https://github.com/linkerd/linkerd2/pull/5144/files#diff-93fa615396627ac17467d427fada1396fee6a35db7184df1723c6d6abb64eab3L77) made to this file that affects the output.

It looks like it might be the `toString` condition and if I understand the `helm ternary` function [docs](https://helm.sh/docs/chart_template_guide/function_list/#ternary), this means that 

```go
(eq (toString .Values.global.proxy.component) "linkerd-identity")
```

returns true, whereas

```go
(eq .Values.global.proxy.component "linkerd-identity")
```

returns false

Did you set a custom namespace or cluster domain when you installed Linkerd?

**Reply by rnsv** [reply]
I am using custom namespace

**Comment by cpretzer** [comment]
thanks @rnsv do you have any other customizations? It will be helpful to see the value of the `.Values.global.proxy.component` value.
